app/commands/markdown/render_html.rb

```diff
@@ -18,13 +18,26 @@ def initialize(options:, nofollow_links: false)
     attr_reader :nofollow_links
 
     def link(node)
-      out('<a href="', node.url.nil? ? '' : escape_href(node.url), '" target="_blank"')
+      out('<a href="', node.url.nil? ? '' : escape_href(node.url), '"')
       out(' title="', escape_html(node.title), '"') if node.title.present?
+      if external_url(node.url)
+        out(' target="_blank"')
+        out(' rel="noopener"') unless nofollow_links
+      end
       out(' rel="nofollow"') if nofollow_links
       out(link_tooltip_attributes(node))
       out('>', :children, '</a>')
     end
 
+    def external_url(url)
+      uri = URI.parse(URI.escape(url)) # rubocop:disable Lint/UriEscapeUnescape
```

Comment:
The `URI.escape` was needed as I otherwise got an error for the Unicode characters in some of the test URL:

> URI::InvalidURIError: URI must be ascii only "https://exercism.io/tracks/ruby/concepts/1_2-3_\xF0\x9F\x8E\xB8"

Unfortunately, `URI.escape` is deprecated:

> W: 33: 23: Lint/UriEscapeUnescape: URI.encode method is obsolete and should not be used. Instead, use CGI.escape, URI.encode_www_form or URI.encode_www_form_component depending on your specific use case.

However, I don't think those alternatives are what we want, as they also escape the `:` in `http://` and thus the URI is not parsed correctly anymore. The alternatives are meant for encoding things in the query string, not the URL itself. There is an `Addressable` gem that was suggested here: https://stackoverflow.com/a/13059657/2071395